lib/FilesHooks.php

```diff
@@ -396,7 +396,16 @@ protected function fileMoving($oldPath, $newPath) {
 		}
 		$accessList = $this->getUserPathsFromPath($parentPath, $parentOwner);
 		$affectedUsers = $accessList['users'];
-		$oldUsers = $this->oldAccessList['users'];
+		$oldAccessList = $this->oldAccessList;
+		if($oldAccessList === null) {
+			$this->logger->warning("Something fishy happens: function fileMove was not finished: {oldPath} -> {path} : {uidOwner}",
```

Comment:
As most returns from `fileMove` are legitimate and not fishy, I think I would not add a log here and rather add some in the `fileMove` move early returns that are unexpected. And I would also be more explicit in the message.
Sorry for the numerous back and forth, and thanks again for addressing the issue :).

Reply:
> As most returns from fileMove are legitimate and not fishy, I think I would not add a log here and rather add some in the fileMove move early returns that are unexpected. 

Yeah, good point.

The state `$oldAccessList === null` is illegal (unhealthy state). It means, that the `FilesHooks::fileMove`  function gets ended unexpected.


I count 4 return points in `FilesHooks::fileMove` :
1) https://github.com/nextcloud/activity/blob/70362c1c0d8d949d6af8f587a8e3409ea3c9265b/lib/FilesHooks.php#L218-L219
2) https://github.com/nextcloud/activity/blob/70362c1c0d8d949d6af8f587a8e3409ea3c9265b/lib/FilesHooks.php#L233-L234
3) https://github.com/nextcloud/activity/blob/70362c1c0d8d949d6af8f587a8e3409ea3c9265b/lib/FilesHooks.php#L273-L274
4) https://github.com/nextcloud/activity/blob/70362c1c0d8d949d6af8f587a8e3409ea3c9265b/lib/FilesHooks.php#L284-L286

And only the 4. case calls the `FilesHooks::fileMoving` function.

https://github.com/nextcloud/activity/blob/70362c1c0d8d949d6af8f587a8e3409ea3c9265b/lib/FilesHooks.php#L295-L312

A more appropriate way would be to invalidate the post process and let error be error, since it is generated by 3rd party.

Cache the case `$this->moveCase` instead of set it direct:

```diff
+		$this->moveCase = false;
+		$moveCase = false;
		if (strpos($oldDir, $newDir) === 0) {
			/**
			 * a/b/c moved to a/c
			 *
			 * Cases:
			 * - a/b/c shared: no visible change
			 * - a/b/ shared: delete
			 * - a/ shared: move/rename
			 */
-			$this->moveCase = 'moveUp';
+			$moveCase = 'moveUp';
		} elseif (strpos($newDir, $oldDir) === 0) {
			/**
			 * a/b moved to a/c/b
			 *
			 * Cases:
			 * - a/b shared: no visible change
			 * - a/c/ shared: add
			 * - a/ shared: move/rename
			 */
-			$this->moveCase = 'moveDown';
+			$moveCase = 'moveDown';
		} else {
			/**
			 * a/b/c moved to a/d/c
			 *
			 * Cases:
			 * - a/b/c shared: no visible change
			 * - a/b/ shared: delete
			 * - a/d/ shared: add
			 * - a/ shared: move/rename
			 */
-			$this->moveCase = 'moveCross';
+			$moveCase = 'moveCross';
		}
```

And set de case at the end to guarantee the availability of `$this->oldAccessList `

```diff
		}

+		$this->moveCase = $moveCase;
		$this->oldAccessList = $oldAccessList;
	}
```

So I would be able to remove the `null` check and logging because it is guaranteed

```diff
		$affectedUsers = $accessList['users'];
-		$oldAccessList = $this->oldAccessList;
-		if($oldAccessList === null) {
-			$this->logger->warning("Something fishy happens: function fileMove was not finished: {oldPath} -> {path} : {uidOwner}",
-				['oldPath'=>$oldPath, 'path'=> $parentPath, 'uidOwner'=>$parentOwner]);
-			$oldAccessList = [
-				'users' => [],
-				'remotes' => [],
-			];
-		}
-		$oldUsers = $oldAccessList['users'];
+		$oldUsers = $this->oldAccessList['users'];

	}
```